### ISSUE TITLE
The comments mention a type void that does not exist

### ISSUE THREAD

**Issue by xavierm02** [math-comp/math-comp#225]
The comments mention a type `void` that does not seem to exist: https://github.com/math-comp/math-comp/search?q=void&unscoped_q=void

**Comment by amahboubi** [comment]
Hi @xavierm02, well spotted. This header is pretty old... May be almost as old as @ggonthier's Four Colour theorem proof?  I guess `void` refers to `False` here. What is your use case?

**Comment by xavierm02** [comment]
I use a type as a flag to (dis)activate a constructor in a inductive:

```
Require Import ssreflect.

Inductive void : Type := .

Definition bool_to_Type b := if b then unit else void.

Section MyType.

  Context {has_B : bool}.

  Inductive my_type :=
    | A : my_type
    | B : bool_to_Type has_B -> my_type
  .
End MyType.

Lemma test1 : forall x : @my_type false, x = A.
  by case.
Qed.

Lemma test2 : exists x y : @my_type true, x <> y.
  exists A.
  exists (@B true tt).
  move => H.
  by inversion H.
Qed.
```

**Comment by CohenCyril** [comment]
Hi, I believe you can do the same by taking `void` to be `False`.

Here is your code with this approach.
To suggest an alternative, I refactored your code a little bit and I also replaced your explicit coercion  `bool_to_Type` by the coercion `is_true` from `bool` to `Prop` and it should be equivalent, since `unit` is equivalent to `true = true`.

```
Require Import ssreflect ssrfun ssrbool.

Inductive my_type (has_B : bool) :=
| A : my_type has_B
| B : has_B -> my_type has_B.
Arguments A {has_B}.
Arguments B {has_B}.

Lemma test1 : forall x : my_type false, x = A.
Proof. by case. Qed.

Lemma test2 : exists x y : my_type true, x <> y.
Proof. by exists A, (B isT) => []. Qed.

```

**Comment by arthuraa** [comment]
FYI I have put up a [repository](https://github.com/arthuraa/coq-void) with instances of basic MathComp classes for `Empty_set`. I might add that to the OPAM archive later, but it would be really useful if those were declared in ssreflect.

**Comment by amahboubi** [comment]
Thanks @arthuraa ! I am adding this to the agenda of the next dev meeting (Oct 16th), so that we provide a feedback by then at the latest.

**Comment by anton-trunov** [comment]
Indeed, including @arthuraa's instances into ssreflect would be great. We also have an instance of `Empty_set` for `eqType` in fcsl-pcm [(here)](https://github.com/imdea-software/fcsl-pcm/blob/b120719fbbfe355044dab4cc104e7bddcb3d7bac/pcm/prelude.v#L241) for precisely the same purpose of "switching off" a constructor.

**Comment by arthuraa** [comment]
I have just added `coq-void` to OPAM (https://github.com/arthuraa/coq-void).

**Comment by ybertot** [comment]
This issue appears to be fixed by #393
